mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1719,17 +1719,16 @@ def set_logged_model_tags(self, model_id: str, tags: list[LoggedModelTag]) -> No
             if not logged_model:
                 self._raise_model_not_found(model_id)
 
-            session.merge(
-                *(
+            # TODO: Consider unserting tags in a single transaction for performance
```

Comment:
https://docs.sqlalchemy.org/en/20/orm/queryguide/dml.html#orm-upsert-statements describes how to bulk-upsert.

Reply:
SQLite implementation would look like:

```python
            from sqlalchemy.dialects.sqlite import insert as sqlite_upsert

            stmt = sqlite_upsert(SqlLoggedModelTag).values(
                [
                    {
                        "model_id": model_id,
                        "experiment_id": logged_model.experiment_id,
                        "tag_key": tag.key,
                        "tag_value": tag.value,
                    }
                    for tag in tags
                ],
            )
            stmt = stmt.on_conflict_do_update(
                index_elements=["model_id", "tag_key"],
                set_={"tag_value": stmt.excluded.tag_value},
            )
            session.execute(stmt)
```